### ISSUE TITLE
Truncated Iptc Tag Descriptions in exiv2 documentation/website

### ISSUE THREAD

**Issue by postscript-dev** [Exiv2/exiv2#1393]
**Describe the bug**
Some of the [Exiv2 - Iptc metadata](https://www.exiv2.org/iptc.html) tag descriptions are truncated compared to the text in [datasets.cpp](../blob/master/src/datasets.cpp).

e.g. In the documentation/website, Iptc.Envelope.ModelVersion Tag Description is:
> A binary number identifying the version of the Information Interchange Model

but inside datasets.cpp
> A binary number identifying the version of the Information Interchange Model, Part I, utilised by the provider. Version numbers are assigned by IPTC and NAA organizations.

**To Reproduce**
NONE.

**Expected behavior**
Documentation/website Iptc tag descriptions to be the same as strings in datasets.cpp .

**Desktop (please complete the following information):**
Website.

**Additional context**
All of the truncated tag descriptions contain a ',' and when the documentation is built, the ',' inside the description string is being used as a separator.

> $ taglist [SectionName] | awk -f iptc.awk > iptc.xml

The taglist program correctly extracts the full tag description but does so without adding "" or some other means of grouping the description text - taglist outputs a csv file.  Awk runs [iptc.awk](../blob/master/doc/templates/iptc.awk) and uses ',' as a separator and this truncates the description which is then displayed in the documentation/website.

**Comment by clanmills** [comment]
Thank You for reporting this.  Before I discuss this issue, may I ask about your handle?  I worked on PostScript for 8 years.  I wrote the AgfaScript PostScript clone 1992-1995 and then at Adobe in the Print Group 1998-2003.

You have eagle eyes to spot this.  Your analysis is totally correct.   The web pages are generated by scripts which use the output of `taglist`.  So:

```bash
606 rmills@rmillsmbp:~/gnu/github/exiv2/0.27-maintenance $ taglist Iptc | grep EnvelopeNumber
EnvelopeNumber, 40, 0x0028, Envelope, true, false, 8, 8, Iptc.Envelope.EnvelopeNumber, String, The characters
form a number that will be unique for the date specified in <DateSent> tag
and for the Service Identifier specified by <ServiceIdentifier> tag. If identical
envelope numbers appear with the same date and with the same Service Identifier, records 2-9 
must be unchanged from the original.
This is not intended to be a sequential serial number reception check.
607 rmills@rmillsmbp:~/gnu/github/exiv2/0.27-maintenance $ 
```

And this has been truncated at the comma and therefore arrives on the web at https://www.exiv2.org/iptc.html as:

```bash
The characters form a number that will be unique for the date specified in <DateSent> 
tag and for the Service Identifier specified by <ServiceIdentifier> tag. 
If identical envelope numbers appear with the same date and with the same Service Identifier
```

We keep the web scripts here:  svn://dev.exiv2.org/svn/team/website.  I'll investigate and fix this in the next day or two and update the web page.

**Comment by postscript-dev** [comment]
Thank you for the quick response.

My handle is postscript-dev because my initials are PS. When creating an account, I couldn't find a suitable ps name combination so went with postscript.  I didn't realise this would cause confusion.

**Comment by clanmills** [comment]
Oh, it's not confusing. I worked with a Paul S, who was never called PostScript!  Actually, I wondered if you were another retired Adobe Engineer in which case we would know each other.

I believe I have this fixed on my machine.  This issue also impacts the webpages concerning XMP properties and the Exif Tags.  I've fixed them also.  I'll update the web-site later today.  

It's amazing what lies at the bottom of the swamp.  Andreas wrote that awk code in 2004 and presumably didn't notice what you spotted.  I've done about 6 releases of Exiv2 and never noticed either.  Mind you, I only glanced at the pages to make sure they weren't insane and have never looked at them critically.  Now they have "commas".  I wouldn't be surprised to learn there was something else "not quite right".

**Comment by clanmills** [comment]
There is another problem in the documentation.  We have tags with descriptions like this:

```cpp
Indicates the unit used to express the GPS receiver speed of movement. "K","M" and "N" represents kilometers per hour, miles per hour, and knots.
```

I changed them all (about 100 or so) to be `'K','M','N'...`.  That's not a good approach.  It will hurt our localisation and I don't want to manually update 100 entries in 18 languages.  I'm going to backtrack and keep the existing descriptions  The correct syntax for this in CSV (as parsed by python csv module) is:

```csv
GPSInfo.GPSSpeedRef,12,0x000c,GPSInfo,Exif.GPSInfo.GPSSpeedRef,Ascii,"Indicates the unit used to express the GPS receiver speed of movement. ""K"",""M"" and ""N"" represents kilometers per hour, miles per hour, and knots."
```

```bash
922 rmills@rmillsmm-local:~/gnu/github/exiv2/0.27-maintenance/doc/templates $ cat tag.txt | python3 tags.py
<?xml version = '1.0'?>
<?xml-stylesheet type="text/xsl" href="tags.xsl"?>
<TAGLIST>
<HEADER>
<title>XYZ MakerNote Tags defined in Exiv2</title>
<text>
<p>Tags found in the MakerNote of images taken with XYZ cameras. These tags 
are defined by Exiv2 in accordance with <a href="makernote.html#RX">[X]</a>.</p>
<p>Click on a column header to sort the table.</p>
</text>
</HEADER>
<ROWSET>

   <ROW num="0">
      <tagname>GPSInfo.GPSSpeedRef</tagname>
      <tagdec>12</tagdec>
      <taghex>0x000c</taghex>
      <ifd>GPSInfo</ifd>
      <key>Exif.GPSInfo.GPSSpeedRef</key>
      <type>Ascii</type>
      <tagdesc>Indicates the unit used to express the GPS receiver speed of movement. "K","M" and "N" represents kilometers per hour, miles per hour, and knots.</tagdesc>
   </ROW>
</ROWSET>
</TAGLIST>
923 rmills@rmillsmm-local:~/gnu/github/exiv2/0.27-maintenance/doc/templates $ 
```

Tomorrow is another day!

**Comment by clanmills** [comment]
Good News.  I've update web 46 pages of documentation on exiv2.org.  Please inspect and let me know about any errors (there will be something).  When you're happy, I'll submit the PR to the code base and we'll be done on this one.
 
<img width="1182" alt="screenshot_33" src="https://user-images.githubusercontent.com/529982/99414126-6b275780-28ee-11eb-9ed5-e020d8991304.png">

**Comment by clanmills** [comment]
There are a couple of typos reported in #1344.  I'll fold them in with the PR for this issue.

**Comment by clanmills** [comment]
Fix submitted: #1400.
